helper/convert/value_test.go

```diff
@@ -0,0 +1,420 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package convert
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/hashicorp/go-cty/cty"
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
+)
+
+func TestPrimitiveTfType(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		Value cty.Value
+		Want  tftypes.Value
+	}{
+		{
+			Value: cty.StringVal("test"),
+			Want:  tftypes.NewValue(tftypes.String, "test"),
+		},
+		{
+			Value: cty.BoolVal(true),
+			Want:  tftypes.NewValue(tftypes.Bool, true),
+		},
+		{
+			Value: cty.NumberIntVal(42),
+			Want:  tftypes.NewValue(tftypes.Number, 42),
+		},
+		{
+			Value: cty.NumberFloatVal(3.14),
+			Want:  tftypes.NewValue(tftypes.Number, 3.14),
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.Value.GoString(), func(t *testing.T) {
+			t.Parallel()
+
+			got := PrimitiveTfValue(test.Value)
```

Comment:
nit: I think for tests like this, you can probably just call the top-level `ToTfValue` since it's how all consumers would call it anyways, plus there isn't a ton of setup needed to call that function anyways.